src/garage/experiment/experiment.py

```diff
@@ -318,3 +201,279 @@ def to_local_command(params,
+        """Make a context from the template information and variant args.
+
+        Currently, all arguments should be keyword arguments.
+
+        Args:
+            args (list): Should be empty.
+            kwargs (dict): Keyword arguments for the wrapped function. Will be
+                logged to `variant.json`
+
+        Returns:
+            ExperimentContext: The created experiment context.
+
+        Raises:
+            ValueError: If args is not empty.
+
+        """
+        if args:
+            raise ValueError('garage.experiment currently only supports '
+                             'keyword arguments')
+        log_dir = self.log_dir
+        if log_dir is None:
+            name = self.name
+            if name is None:
+                name = self.function.__name__
+            log_dir = ('{data}/local/{prefix}/{name}'.format(
+                data=osp.join(os.getcwd(), 'data'),
+                prefix=self.prefix,
+                name=name))
+        log_dir = _make_sequential_log_dir(log_dir)
+
+        tabular_log_file = os.path.join(log_dir, 'progress.csv')
+        text_log_file = os.path.join(log_dir, 'debug.log')
+        variant_log_file = os.path.join(log_dir, 'variant.json')
+
+        dump_json(variant_log_file, kwargs)
+
+        logger.add_output(dowel.TextOutput(text_log_file))
+        logger.add_output(dowel.CsvOutput(tabular_log_file))
+        logger.add_output(dowel.TensorBoardOutput(log_dir))
+        logger.add_output(dowel.StdOutput())
+
+        logger.push_prefix('[%s] ' % self.name)
+
+        return ExperimentContext(snapshot_dir=log_dir,
+                                 snapshot_mode=self.snapshot_mode,
+                                 snapshot_gap=self.snapshot_gap)
+
+    def __call__(self, *args, **kwargs):
+        """Wrap a function to turn it into an ExperimentTemplate.
+
+        Note that this docstring will be overriden to match the function's
+        docstring on the ExperimentTemplate once a function is passed in.
+
+        Args:
+            args (list): If no function has been set yet, must be a list
+                containing a single callable.
+            kwargs (dict): Arguments passed onto the wrapped function.
+
+        Returns:
+            object: The returned value of the wrapped function.
+
+        Raises:
+            ValueError: If not passed a single callable argument.
+
+        """
+        if self.function is None:
+            if len(args) != 1 or len(kwargs) != 0 or not callable(args[0]):
+                raise ValueError('Please apply the result of '
+                                 'wrap_experiment() to a single function')
+            # Apply ourselves as a decorator
+            self.function = args[0]
+            functools.update_wrapper(self, self.function)
+            return self
+        else:
+            ctxt = self._make_context(*args, **kwargs)
+            return self.function(ctxt, *args, **kwargs)
+
+
+def wrap_experiment(function=None,
```

Comment:
in the spirit of most python decorators, shall we just call this `@experiment`?

```python
from garage.experiment import experiment

@experiment
def my_cool_exp(ctxt):
   # cool stuff here

@experiment
def my_cool_exp2(ctxt):
    # even cooler stuff here
```

Reply:
I would like to, but moving the existing `garage.experiment` module would take significant work right now.